src/SingleArticle.jsx

```diff
@@ -15,8 +16,39 @@ export const SingleArticle = () => {
     });
   }, [article_id]);
 
+  const upVote = () => {
+    setArticle((currentArticle) => {
+      return { ...currentArticle, votes: currentArticle.votes + 1 };
+    });
+    const updatedVotes = {
+      inc_votes: 1,
+    };
+    patchArticle(article_id, updatedVotes).catch((error) => {
+      setError(error.response);
+      setArticle((currentArticle) => {
+        return { ...currentArticle, votes: currentArticle.votes - 1 };
+      });
+    });
+  };
+
+  const downVote = () => {
+    setArticle((currentArticle) => {
+      return { ...currentArticle, votes: currentArticle.votes - 1 };
+    });
+    const updatedVotes = {
+      inc_votes: -1,
+    };
+    patchArticle(article_id, updatedVotes).catch((error) => {
+      setError(error.response);
+      setArticle((currentArticle) => {
+        return { ...currentArticle, votes: currentArticle.votes + 1 };
+      });
+    });
+  };
   if (isLoading) {
-    return <h2>Loading Articles Please Wait...</h2>;
+    return <h2>Loading Article Please Wait...</h2>;
```

Comment:
I wouldn't expect an error message to remove all the page content. If the patch request for voting were to fail, a smaller error message under the voting buttons would be better